core/src/main/java/org/mule/runtime/core/internal/processor/AsyncDelegateMessageProcessor.java

```diff
@@ -293,4 +309,57 @@ public void setMaxConcurrency(Integer maxConcurrency) {
     // Lifecycle of inner objects is already handled by this class' lifecycle methods
     return emptyList();
   }
+
+  private static class QueueBackpressureHandler implements Startable, Stoppable {
+
+    private final BlockingQueue<CoreEvent> asyncQueue;
+    private final SchedulerService schedulerService;
+    private final Supplier<SchedulerConfig> schedulerConfigSupplier;
+    private final String location;
+    private final Consumer<CoreEvent> eventDispatcher;
+
+    private Scheduler queueDispatcherScheduler;
+
+    public QueueBackpressureHandler(SchedulerService schedulerService, Supplier<SchedulerConfig> schedulerConfigSupplier,
+                                    Consumer<CoreEvent> eventDispatcher, String location) {
+      this.schedulerService = schedulerService;
+      this.schedulerConfigSupplier = schedulerConfigSupplier;
+
+      this.asyncQueue = new LinkedBlockingQueue<>();
+      this.eventDispatcher = eventDispatcher;
+
+      this.location = location;
+    }
+
+    public void handleBackpressure(CoreEvent event) {
+      asyncQueue.offer(event);
+    }
+
+    @Override
+    public void start() {
+      final SchedulerConfig schedulerConfig = schedulerConfigSupplier.get().withName(location + " - queue dispatcher");
+      queueDispatcherScheduler = schedulerService.ioScheduler(schedulerConfig);
+
+      queueDispatcherScheduler.scheduleWithFixedDelay(() -> {
+        try {
+          final CoreEvent event = asyncQueue.peek();
+          if (event != null) {
+            eventDispatcher.accept(event);
+            asyncQueue.remove(event);
+          }
+        } catch (FromFlowRejectedExecutionException free) {
+          // nothing to do, a retry will come next
+        }
+      }, 0, 2, MILLISECONDS);
```

Comment:
isn't this too much frequency? 

Also:
a) isn't a busy wait preferable in cases of high throughput
b) isn't a wait()/notify() approach preferable in most cases?